distmap

```diff
@@ -235,31 +241,49 @@ GetOptions(
 	"disable-zipped-output" 		=>\$nozip,
 	"hadoop-home=s" 			=>\$hadoop_home,
 	"mappers-exe-archive=s" 		=>\$mappers_exe_archive,
-	
-	
-	"mapper=s" 				=> \@mapper,		# to check	
+
+
+	"mapper=s" 				=> \@mapper,		# to check
 	"mapper-path=s"				=> \@mapper_path,
 	"picard-jar=s"				=> \$picard_jar,
 	"picard-mergesamfiles-jar=s"		=> \$picard_mergesamfiles_jar,	# to ckeck
 	"picard-sortsam-jar=s"			=> \$picard_sortsam_jar,	# to check
+	"readtools=s"         => \$readtools,
 	"mapper-args=s"				=> \@mapper_args,
 	"bwa-sampe-args=s"			=> \$bwa_sampe_args,
-	"output-format=s"			=> \$output_format,					
+	"output-format=s"			=> \$output_format,
 	"job-desc=s"				=> \$job_desc,	# to check
 	"queue-name=s"				=> \$queue_name,	# to check
 	"gsnap-output-split"			=> \$gsnap_output_split,	# to check
 	"verbose"        		        => \$verbose,
 	"help"          		        =>\$help,
 	"h"          		        	=>\$help
-) or pod2usage(-msg=>"Wrong options",-verbose=>1); 
+) or pod2usage(-msg=>"Wrong options",-verbose=>1);
 
 pod2usage(-verbose=>2) if $help;
 
+if ( $hadoop_home ne "" ) {
+    # if $hadoop_home is set, make sure there is no trailing slash
+    $hadoop_home =~ s/\/$//;
+}
+else {
+    # try to find "hadoop" in path and prune "/bin/hadoop" from the full path
+    $hadoop_home = `which hadoop`;
+    $hadoop_home =~ s/\/bin\/hadoop$//;
+}
+
+if ( $readtools =~ /\.jar$/ ) {
+  $readtools = "java -Xmx8g -Dsnappy.disable=true -jar $readtools";
+}
+unless ( `$readtools --version` gt "1.1.0" ) {
```

Comment:
Will this work with a developmental version higher than "1.1.0". It might be useful for me to test if nothing is broken while evolving ReadTools.

Reply:
I guess this depends on how you name these versions. "true" are, for example,

 "2.0" gt "1.0"
 "1.1.0" gt "1.1-devel"
 "1.2-release" gt "1.2-devel"
but also 
 "2.0.1-devel" gt "2.0.1"

So if one assigns patch numbers only upon release, it should be safe?

Reply:
Releases are following SemVer (from the final user point of view, the java API isn't stable yet), so they are always expected to be "1.1.0", "1.1.1", etc.

Nevertheless, developmental versions are using git-tags, which looks like `1.1.0-6-gb23ebb8-SNAPSHOT` and means:

* 1.1.0 = latest version it comes from
* 6 = number of commits since the latest version (in this case, a branch with 6 new commits).
* gb23ebb8 = commit number
* SNAPSHOT = indication that this is not a release

Thus, the requirement here is the following:

* `"1.1.0-6-gb23ebb8-SNAPSHOT" >= "1.1.0" == true`
* `"1.1.1-6-gb23ebb8-SNAPSHOT" >= "1.1.0" == true`
* `"1.0.0-6-gb23ebb8-SNAPSHOT" >= "1.1.0" == false`

If this is working, then it should be fine as it is now...